controllers/cluster_controller.go

```diff
@@ -435,9 +435,44 @@ func (r *FoundationDBClusterReconciler) updateOrApply(ctx context.Context, clust
 	return r.Status().Update(ctx, cluster)
 }
 
+// getAllPodStatus will get the current status and phase of all the pods in the fdb cluster and check if all of them are up before proceeding with
+// connecting to fdb cluster and get information on coordinator servers, etc. Once all pods are up and running it will return true else false.
+func (r *FoundationDBClusterReconciler) getAllPodStatus(ctx context.Context, logger logr.Logger, cluster *fdbv1beta2.FoundationDBCluster) bool {
```

Comment:
What is the benefit of waiting for all Pods to be up and running? For a huge cluster this could take a long time until all Pods are scheduled and running.

Reply:
@johscheuer Won't the cluster still be in an unhealthy state without all the log and storage servers pods up and running?
For both small or large cluster if the Storage Server pods are not up and healthy, cluster might throw unhealthy state with no data replica error.

Reply:
What do you mean by an "unhealthy" state? The cluster won't show the reconciled state until all Pods are up and running.

By recovered one you mean a cluster where you deleted all Pods and waited until the operator recreated them?